pkg/leeway/build.go

```diff
@@ -390,39 +386,53 @@
 	}
 
 	requirements := pkg.GetTransitiveDependencies()
+
 	allpkg := append(requirements, pkg)
 
-	// respect per-package cache level when downloading from remote cache
-	remotelyCachedReq := make([]*Package, 0, len(requirements))
-	remotelyCachedReq = append(remotelyCachedReq, requirements...)
+	pkgsInLocalCache := make(map[*Package]bool)
```

Comment:
A `map[*Package]...` is a map of pointers (memory address), so it's not really an ideal way to compare if something exists or not as it can lead to some weird cases (content is not compared). I would personally use a `map[string]struct{}` with the package name as the key. In this case it might be somewhat okay, though, but I haven't looked at how the Package is being used throughout the codebase